deluge/ui/web/json_api.py

```diff
@@ -141,8 +142,12 @@ def enable(self):
                 self.connect()
 
     def _on_client_disconnect(self, *args):
-        component.get("Web.PluginManager").stop()
-        component.get("Web").stop()
+        try:
+            component.get("Web.PluginManager").stop()
+            component.get("Web").stop()
+        except KeyError:
+            # Unit tests don't have these components
```

Comment:
I think this would be better caught in the test with a get() method as per https://github.com/deluge-torrent/deluge/blob/develop/deluge/tests/test_torrent.py#L27